test/common/upstream/health_checker_impl_test.cc

```diff
@@ -138,13 +138,16 @@ class HttpHealthCheckerImplTest : public testing::Test {
     unhealthy_threshold: 2
     healthy_threshold: 2
     http_health_check:
-      service_name: locations
+      service_name_matcher:
+        prefix: locations
       path: /healthcheck
       codec_client_type: Http2
     )EOF";
 
-    health_checker_.reset(new TestHttpHealthCheckerImpl(*cluster_, parseHealthCheckFromV2Yaml(yaml),
```

Comment:
Apologies, what I meant here is to modify `parseHealthCheckFromV2Yaml` to force validation in that function. There is no reason not to validate in all of the tests. (It's possible you might need to fix some other tests, I'm not sure).

Reply:
Thanks @mattklein123 , it makes sense. I've Added the validation inside `parseHealthCheckFromV2Yaml` now.